elections/uk/management/commands/uk_create_elections_from_every_election.py

```diff
@@ -0,0 +1,193 @@
+            if election.get('election_id'):
+                if election.get('group_type') == "organisation":
+                    self.process_election(election)
+
+    def process_election(self, election_dict):
+        """
+        Create the elections
+        """
+        print(election_dict['election_id'])
+        election_model = self.get_election(election_dict)
+
+        organization_extra = self.get_or_create_organisation(
+            election_model, election_dict)
+
+        if election_dict['children']:
+            for division_id in election_dict['children']:
+                self.add_single_division(
+                    election_model, organization_extra, division_id)
+        else:
+            # This is an election to the org directly, with no divisions
+            # e.g. a mayor or PCC
+            self.add_area(
+                election_dict, election_model, organization_extra)
+
+    def get_election(self, election_dict):
+        election_id = election_dict['election_id']
+        election_date = election_dict['poll_open_date']
+
+        # TODO Set current from the API
+        current = True
+
+        return Election.objects.update_or_create(
+            slug=election_id,
+            election_date=election_date,
+            current=current,
+            defaults={
+                "name": election_dict['election_title'],
+                "for_post_role": election_dict['elected_role'],
+                "candidate_membership_role": "Candidate",
+                "show_official_documents": True,
+            }
+        )[0]
+
+    def add_single_division(self, election_model,
+                            organization_extra, division_id):
+        url = "{}api/elections/{}".format(
+            self.EE_BASE_URL, division_id)
+        area_info = requests.get(url).json()
+
+        self.add_area(area_info, election_model, organization_extra)
+
+    def get_or_create_organisation(self, election_model, election_dict):
+        org_name = election_dict['organisation']['official_name']
+        org_slug = election_dict['organisation']['slug']
+        classification = election_dict['organisation']['organisation_type']
+
+        try:
+            organization_extra = OrganizationExtra.objects.get(
+                slug=org_slug)
+            organization = organization_extra.base
+        except OrganizationExtra.DoesNotExist:
+            organization, _ = Organization.objects.get_or_create(
+                name=org_name,
+                defaults={'classification': classification})
+            organization_extra = OrganizationExtra.objects.create(
+                base=organization,
+                slug=org_slug
+            )
+        if not election_model.organization:
+            election_model.organization = organization_extra.base
+            election_model.save()
+        return organization_extra
+
+    def add_area(self, area_info, election, organization_extra):
```

Comment:
I found this method quite hard to follow compared to the others, at least partly because it's quite long. Some of it could be factored out into methods (e.g. `get_party_set(territory_code)` for the first lines, or `get_area_type` for the next).

The bits that are based on `area_info['division']` could maybe do with a comment or two about what it's doing? Since I'm not very familiar with what these values from the API mean, that would help me anyway. (I couldn't find documentation of them easily on https://elections.democracyclub.org.uk but if there's somewhere that could be linked to, that might help too.)